packages/medusa/src/workflows/admin/create-products/definition.ts

```diff
@@ -0,0 +1,421 @@
+import {
+  TransactionHandlerType,
+  TransactionPayload,
+  TransactionStepHandler,
+  TransactionStepsDefinition,
+} from "../../../utils/transaction"
+import {
+  IInventoryService,
+  MedusaContainer,
+  ProductTypes,
+} from "@medusajs/types"
+import {
+  defaultAdminProductFields,
+  defaultAdminProductRelations,
+} from "../../../api"
+import {
+  attachInventoryItems,
+  attachSalesChannelToProducts,
+  attachShippingProfileToProducts,
+  createInventoryItems,
+  createProducts,
+  CreateProductsData,
+  CreateProductsPreparedData,
+  detachInventoryItems,
+  detachSalesChannelFromProducts,
+  detachShippingProfileFromProducts,
+  prepareCreateProductsData,
+  removeInventoryItems,
+  removeProducts,
+  updateProductsVariantsPrices,
+} from "../../functions"
+import { PricingService, ProductService } from "../../../services"
+import { CreateProductsWorkflowInputData, InjectedDependencies } from "./types"
+
+export enum CreateProductsWorkflowActions {
+  prepare = "prepare",
+  createProducts = "createProducts",
+  attachToSalesChannel = "attachToSalesChannel",
+  attachShippingProfile = "attachShippingProfile",
+  createPrices = "createPrices",
+  createInventoryItems = "createInventoryItems",
+  attachInventoryItems = "attachInventoryItems",
+  result = "result",
+}
+
+export const workflowSteps: TransactionStepsDefinition = {
+  next: {
+    action: CreateProductsWorkflowActions.prepare,
+    noCompensation: true,
+    next: {
+      action: CreateProductsWorkflowActions.createProducts,
+      next: [
+        {
+          action: CreateProductsWorkflowActions.attachShippingProfile,
+          saveResponse: false,
+        },
+        {
+          action: CreateProductsWorkflowActions.attachToSalesChannel,
+          saveResponse: false,
+        },
+        {
+          action: CreateProductsWorkflowActions.createPrices,
+          next: {
+            action: CreateProductsWorkflowActions.createInventoryItems,
+            next: {
+              action: CreateProductsWorkflowActions.attachInventoryItems,
+              next: {
+                action: CreateProductsWorkflowActions.result,
+                noCompensation: true,
+              },
+            },
+          },
+        },
+      ],
+    },
+  },
+}
+
+const shouldSkipInventoryStep = (
```

Comment:
suggestion: move to `/functions` ?

Reply:
I ve put it here as an helper, but if a user add another step related to the inventory, he might want to skip in other circumstences and I didn't want to confuse them, wdyt?

Reply:
Got it. Fine to keep as is